File: easy/python/ransom_note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        char_count = {}
        count = 0

        for char in ransomNote:
            if char in char_count:
                char_count[char] += 1
            else:
                char_count[char] = 1
            count += 1

        for char in magazine:
            if char in char_count and char_count[char] > 0:
                char_count[char] -= 1
                count -= 1

        return count == 0

File: easy/ruby/ransom_note.rb
# @param {String} ransom_note
# @param {String} magazine
# @return {Boolean}
def can_construct(ransom_note, magazine)
  hash = Hash.new(0)
  count = 0
  ransom_note.each_char do |char|
    hash[char] = (hash[char] || 0) + 1
    count += 1
  end

  magazine.each_char do |char|
    next unless hash[char].positive?

    hash[char] -= 1
    count -= 1
  end

  count.zero?
end
